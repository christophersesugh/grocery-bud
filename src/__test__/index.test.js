import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Tab from "../Tab";

afterEach(() => {
  cleanup();
});

test("should render non-completed todo", () => {
  const todo = {
    id: 1,
    name: "item 1",
    completed: false,
  };
  render(<Tab todo={todo} />);
  const tabElement = screen.getByTestId("todo-1");
  expect(tabElement).toBeInTheDocument();
  expect(tabElement).toHaveTextContent("item 1");
  expect(tabElement).not.toContainHTML("<strike>");
});

test("matches snapshot", () => {
  const todo = {
    id: 2,
    name: "item 2",
    completed: true,
  };
  const tree = renderer.create(<Tab todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
  console.log(tree);
});
