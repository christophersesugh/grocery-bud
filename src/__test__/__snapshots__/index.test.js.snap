// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="tab"
  data-testid="todo-2"
>
  <h1>
    todo - 
    <strike>
      <h1>
        item 2
      </h1>
    </strike>
  </h1>
</div>
`;
